docs/DeveloperGuide.md

```diff
@@ -1232,12 +1315,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. Nurse's input is invalid command format.
+
+  * 3a1. PatientSync shows an error message.
+
+    Use case resumes at step 2.
+
+* 3b. Nurse's input contains invalid patient index.
+
+  * 3b1. PatientSync shows an error message.
+
+    Use case resumes at step 2.
+
+* 3c. Nurse's input contains invalid tags.
+
+  * 3c1. PatientSync shows an error message.
+
+    Use case resumes at step 2.
+
+**Use case: UC10 - Find patients by a tag**
+
+**MSS**
+
+1. Nurse requests to list patients.
+1. PatientSync shows a list of patients.
+1. Nurse requests to find patients with a specific tag in the list.
+1. PatientSync finds patients with the specified tag and displays success message.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. Nurse's input is invalid command format.
 
     * 3a1. PatientSync shows an error message.
 
       Use case resumes at step 2.
 
+**Use case: UC11 - Find patients by keyword**
```

Comment:
```suggestion
**Use case: UC11 - Find patients by name**
```

Reply:
updated!

Reply:
After offline discussion, it is ok to change to name!